build.gradle.kts

```diff
@@ -18,9 +18,108 @@ java {
+}
+
+tasks.register<Exec>("licenseHeader") {
+    dependsOn("installLicenseHeader")
+    description = "Runs the Buf license-header CLI."
+    commandLine(
+        bufLicenseHeaderCLIPath,
+        "--license-type",
+        "apache",
+        "--copyright-holder",
+        "Buf Technologies, Inc.",
+        "--year-range",
+        project.findProperty("license-header.years")!!.toString(),
+        "--ignore",
+        "src/main/java/build/buf/validate/",
+        "--ignore",
+        "conformance/src/main/java/build/buf/validate/conformance/",
+    )
+}
+
+tasks.register<Exec>("generateTestSourcesImports") {
+    dependsOn("installBuf")
+    description = "Generates code with buf generate --include-imports for unit tests."
+    commandLine(
+        bufCLIPath,
+        "generate",
+        "--template",
+        "src/test/resources/proto/buf.gen.imports.yaml",
+        "src/test/resources/proto",
+        "--include-imports",
+    )
+}
+
+tasks.register<Exec>("generateTestSourcesNoImports") {
+    dependsOn("installBuf")
+    description = "Generates code with buf generate --include-imports for unit tests."
+    commandLine(bufCLIPath, "generate", "--template", "src/test/resources/proto/buf.gen.noimports.yaml", "src/test/resources/proto")
+}
+
+tasks.register("generateTestSources") {
+    dependsOn("generateTestSourcesImports", "generateTestSourcesNoImports")
+    description = "Generates code with buf generate for unit tests"
+}
+
+tasks.register<Exec>("exportProtovalidateModule") {
+    dependsOn("installBuf")
+    description = "Exports the bufbuild/protovalidate module sources to src/main/resources."
+    commandLine(
+        bufCLIPath,
+        "export",
+        "buf.build/bufbuild/protovalidate:${project.findProperty("protovalidate.version")}",
+        "--output",
+        "src/main/resources",
+    )
+}
+
+tasks.register<Exec>("generateSources") {
+    dependsOn("installBuf")
+    description = "Generates sources for the bufbuild/protovalidate module sources to src/main/java."
+    commandLine(bufCLIPath, "generate", "--template", "buf.gen.yaml", "src/main/resources")
+}
+
+tasks.register<Exec>("generateConformance") {
+    dependsOn("installBuf")
+    description = "Generates sources for the bufbuild/protovalidate-testing module to conformance/src/main/java."
+    commandLine(bufCLIPath, "generate", "--template", "buf.gen.yaml", "src/main/resources")
+}
+
+tasks.register("generate") {
+    description = "Generates sources with buf generate and buf export."
+    dependsOn(
+        "generateTestSources",
+        "exportProtovalidateModule",
+        "generateSources",
+        "generateConformance",
+        "licenseHeader",
+    )
+}
+
 tasks.withType<JavaCompile> {
-    if (JavaVersion.current().isJava9Compatible) doFirst {
-        options.compilerArgs = mutableListOf("--release", "8")
+    dependsOn("generateTestSources")
```

Comment:
This ensures that we've generated the source code necessary to compile the tests prior to compiling code.